gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -0,0 +1,124 @@
+        - gnomad 3.1 complex indels: `CADD-1.6-gnomad-complex-variants.ht` was run on gnomAD v3.1 with CADD v1.6 in 2020. It contains 2,307 complex variants that do not fit Hail's criteria for an indel and thus exist in a separate table than the gnomad 3.1 indels.
+        - gnomAD v4 indels: `cadd.v1.6.gnomAD_v4_new_indels.tsv.bgz` (368M) was run on gnomAD v4 with CADD v1.6 in 2023. It contains 32,561,253 indels that are new in gnomAD v4.
+
+         .. note::
+         1,972,208 indels were duplicated in gnomAD v3.0 and v4.0 or in gnomAD v3.1 and v4.0. However, CADD only generates a score per loci. We keep only the latest prediction, v4.0, for these loci.
+    :return: Hail Table with CADD scores for GRCh38.
+    """
+
+    def _load_cadd_raw(cadd_tsv) -> hl.Table:
+        """Functions to load CADD raw data in TSV format to Hail Table."""
+        column_names = {
+            "f0": "chr",
+            "f1": "pos",
+            "f2": "ref",
+            "f3": "alt",
+            "f4": "RawScore",
+            "f5": "PHRED",
+        }
+        types = {"f0": hl.tstr, "f1": hl.tint32, "f4": hl.tfloat32, "f5": hl.tfloat32}
+        ht = hl.import_table(
+            cadd_tsv,
+            types=types,
+            no_header=True,
+            force_bgz=True,
+            comment="#",
+            min_partitions=1000,
+        )
+        ht = ht.rename(column_names)
+        chr = hl.if_else(ht.chr.startswith("chr"), ht.chr, hl.format("chr%s", ht.chr))
+        ht = ht.annotate(
+            locus=hl.locus(chr, ht.pos, reference_genome="GRCh38"),
+            alleles=hl.array([ht.ref, ht.alt]),
+        )
+        ht = ht.select("locus", "alleles", "RawScore", "PHRED")
+        ht = ht.key_by("locus", "alleles")
+
+        return ht
+
+    snvs = _load_cadd_raw(
+        "gs://gnomad-qin/v4_annotations/cadd.v1.6.whole_genome_SNVs.tsv.bgz"
+    )
+    indel3_0 = _load_cadd_raw(
+        "gs://gnomad-qin/v4_annotations/cadd.v1.6.gnomad.genomes.r3.0.indel.tsv.bgz"
+    )
+    indel3_1 = hl.read_table("gs://gnomad-wphu/CADD-indels-gnomad.3.1.ht")
+    indel3_1_complex = hl.read_table(
+        "gs://gnomad-wphu/CADD-1.6-gnomad-complex-variants.ht"
+    )
+    indel4 = _load_cadd_raw(
+        "gs://gnomad-qin/v4_annotations/cadd.v1.6.gnomAD_v4_new_indels.tsv.bgz"
+    )
+
+    # Merge the CADD predictions run for v3 versions.
+    indel3 = indel3_0.union(indel3_1, indel3_1_complex)
+
+    # This will avoid duplicated indels in gnomAD v3 and v4.
+    indel3 = indel3.anti_join(indel4)
+
+    ht = snvs.union(indel3, indel4)
+    ht = ht.annotate_globals(cadd_version="v1.6")
+    return ht
+
+
+def main(args):
+    """Generate Hail Tables with in silico predictors."""
+    hl.init(
+        default_reference="GRCh38",
+        log="/insilico_predictors.log",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+
+    if args.cadd:
+        logger.info("Creating CADD Hail Table for GRCh38...")
+
+        ht = create_cadd_grch38_ht()
+        ht.checkpoint(
```

Comment:
As we discussed in person, this can be changed back to a write.

Reply:
I changed it back to `write`, but it won't have `_read_if_exists` anymore, does it matter?